finality-aleph/src/network/io.rs

```diff
@@ -1,33 +1,42 @@
 use futures::channel::mpsc;
 
 use crate::network::{
-    manager::NetworkData, ConnectionManagerIO, Data, Multiaddress, NetworkServiceIO as NetworkIo,
-    SessionManagerIO,
+    manager::{DataInSession, VersionedAuthentication},
+    ConnectionManagerIO, Data, Multiaddress, NetworkServiceIO as NetworkIO, SessionManagerIO,
 };
 
-type NetworkServiceIO<D, M> = NetworkIo<NetworkData<D, M>, M>;
+type AuthenticationNetworkIO<D, M> = NetworkIO<VersionedAuthentication<M>, DataInSession<D>, M>;
 
 pub fn setup<D: Data, M: Multiaddress + 'static>() -> (
     ConnectionManagerIO<D, M>,
-    NetworkServiceIO<D, M>,
+    AuthenticationNetworkIO<D, M>,
     SessionManagerIO<D>,
 ) {
     // Prepare and start the network
     let (commands_for_network, commands_from_io) = mpsc::unbounded();
+    let (data_for_network, data_from_user) = mpsc::unbounded();
     let (messages_for_network, messages_from_user) = mpsc::unbounded();
```

Comment:
nit: Maybe now this name should indicate it is discovery messages? Not sure if there is any good name for that though. `discovery_for_network`/`authentications_for_network`? `auth_data_for_network` is definitely too long. That said, a lot of places would have to be renamed and just `messages` is close enough, so up to you.

Also you already used `authentications_from_network` in ConnectionManager, so maybe this is the way.

Reply:
Yeah, the reason I did not make this consistently `authentications_` is as in the comment below, I suspect a lot of this will be abstracted away using proper interfaces soon-ish, so adding a lot of changes just to have temporarily simpler names doesn't seem necessary.